File: Src/java/engine/src/test/java/org/opencds/cqf/cql/engine/elm/executing/InValueSetEvaluatorTest.java
package org.opencds.cqf.cql.engine.elm.executing;

import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertInstanceOf;

import org.junit.jupiter.api.Test;
import org.opencds.cqf.cql.engine.execution.Environment;
import org.opencds.cqf.cql.engine.execution.State;
import org.opencds.cqf.cql.engine.runtime.ValueSet;

public class InValueSetEvaluatorTest {

    @Test
    void issue1469FalseOnNullCode() {
        var env = new Environment(null);
        var state = new State(env);
        var valueSet = new ValueSet();

        Object actual = InValueSetEvaluator.inValueSet(null, valueSet, state);
        assertInstanceOf(Boolean.class, actual);
        assertFalse((Boolean) actual);
    }
}

File: Src/java/engine/src/jvmTest/kotlin/org/opencds/cqf/cql/engine/elm/executing/InValueSetEvaluatorTest.kt
package org.opencds.cqf.cql.engine.elm.executing

import kotlin.test.Test
import kotlin.test.assertFalse
import kotlin.test.assertIs
import org.opencds.cqf.cql.engine.execution.Environment
import org.opencds.cqf.cql.engine.execution.State
import org.opencds.cqf.cql.engine.runtime.ValueSet

class InValueSetEvaluatorTest {
    @Test
    fun issue1469FalseOnNullCode() {
        val env = Environment(null)
        val state = State(env)
        val valueSet = ValueSet()

        val actual = InValueSetEvaluator.inValueSet(null, valueSet, state)
        assertIs<Boolean>(actual)
        assertFalse(actual)
    }
}
